### ISSUE TITLE
Feature: Allow Localizations to be filtered by Frame-associated States

### ISSUE THREAD

**Issue by sbatchelder** [cvisionai/tator#1306]
We would like to be able to filter and view localizations in the gallery based on whether they share a media and frame with a Frame-associated state.

**Comment by bctcvai** [comment]
Prior to UI hookup it is advisable to move this search to `object_search`.  Similarly we should create "directionaly aware related search" parameters and deprecate "related_search" as it is too loosely defined in cases when relations are ambiguous. 

This issue is a good example; in this case we are searching on `related_frame_states` so the grammar should support that vs. just `related_search`(which for metadata endpoints means `Media`)  For `Media` endpoints related search means both states and localizations which is more useful in that direction, if not, slightly confusing. 

Lastly, related_search, related_frame_state_search, related_media_state_search,etc... may be worth collapsing into object_search itself versus more PUT body or REST query parameters. This may look like:

```
object_search = {'attribute': '$related_frame_state_search', 'operation': 'eq', 'value': {...recursively defined object search}} 
```

**Comment by bctcvai** [comment]
Closed in #1592 

Localizations in the  gallery can now be filtered by coincident states, or localization that share the same frame as a state.
